src/Spice86/Views/DisassemblyView.axaml.cs

```diff
@@ -1,57 +1,85 @@
+    private IDisassemblyViewModel? _viewModel;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DisassemblyView"/> class.
+    /// </summary>
     public DisassemblyView() {
         InitializeComponent();
-        FunctionComboBox.SelectionChanged += FunctionComboBox_SelectionChanged;
-        StartAddressTextBox.KeyUp += StartAddressTextBox_KeyUp;
-        DisassemblyDataGrid.KeyUp += DisassemblyDataGrid_KeyUp;
-        NumberOfInstructionsShownNumericUpDown.KeyUp += NumberOfInstructionsShownNumericUpDown_KeyUp;
+        DataContextChanged += DisassemblyView_DataContextChanged;
+        
+        // Subscribe to attached/detached events
+        AttachedToVisualTree += DisassemblyView_AttachedToVisualTree;
+        DetachedFromVisualTree += DisassemblyView_DetachedFromVisualTree;
     }
 
-    private void NumberOfInstructionsShownNumericUpDown_KeyUp(object? sender, KeyEventArgs e) {
-        if (DataContext is DisassemblyViewModel viewModel && e.Key == Key.Enter &&
-            viewModel.UpdateDisassemblyCommand.CanExecute(null)) {
-            viewModel.UpdateDisassemblyCommand.Execute(null);
+    private void InitializeComponent() {
+        AvaloniaXamlLoader.Load(this);
+    }
+
+    private void DisassemblyView_DataContextChanged(object? sender, EventArgs e) {
+        // Unsubscribe from the old view model if it exists
+        if (_viewModel != null) {
+            _viewModel.PropertyChanged -= ViewModel_PropertyChanged;
+        }
+
+        // Subscribe to the new view model
+        _viewModel = DataContext as IDisassemblyViewModel;
+        if (_viewModel != null) {
+            _viewModel.PropertyChanged += ViewModel_PropertyChanged;
         }
     }
+    
+    private void DisassemblyView_AttachedToVisualTree(object? sender, Avalonia.VisualTreeAttachmentEventArgs e) {
+        // Activate the view model when the view is attached to the visual tree
+        _viewModel?.Activate();
+    }
+
+    private void DisassemblyView_DetachedFromVisualTree(object? sender, Avalonia.VisualTreeAttachmentEventArgs e) {
+        // Deactivate the view model when the view is detached from the visual tree
+        _viewModel?.Deactivate();
+    }
 
-    private void StartAddressTextBox_KeyUp(object? sender, KeyEventArgs e) {
-        ExecUpdateDisassemblyCommand(e, true);
+    private static void ViewModel_PropertyChanged(object? sender, PropertyChangedEventArgs e) {
     }
 
-    private void ExecUpdateDisassemblyCommand(KeyEventArgs e, bool isUsingLinearAddressing) {
-        if (DataContext is DisassemblyViewModel viewModel && e.Key == Key.Enter &&
-            viewModel.UpdateDisassemblyCommand.CanExecute(null)) {
-            viewModel.UpdateDisassemblyCommand.Execute(null);
+    private void OnBreakpointClicked(object? sender, TappedEventArgs e) {
+        if (sender is Control {DataContext: DebuggerLineViewModel debuggerLine}) {
+            _viewModel?.ToggleBreakpointCommand.Execute(debuggerLine);
```

Comment:
Check for CanExecute first on the ICommand API.

Reply:
Done.